pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -648,19 +647,27 @@ func (p *Pod) validatePodGroupMetadata(r record.EventRecorder, activePods []core
 	return nil
 }
 
-// activePods returns a slice of non-failed pods in the group
+// activePods returns a slice of active pods in the group
 func (p *Pod) activePods() []corev1.Pod {
-	// Filter all failed pods
 	activePods := make([]corev1.Pod, 0, len(p.list.Items))
 	for _, pod := range p.list.Items {
-		if pod.Status.Phase != corev1.PodFailed {
+		if isPodActive(&pod) {
 			activePods = append(activePods, pod)
 		}
 	}
 
 	return activePods
 }
 
+// isPodActive returns whether the Pod can eventually run, is Running or Succeeded.
```

Comment:
I'm wondering if naming "succeeded" pods can be misleading, especially because this term has a different meaning in k8s: https://github.com/kubernetes/kubernetes/blob/4b94168c0f724bbf1edfd846a8e788a32d9a5c49/pkg/controller/controller_utils.go#L980. 

I would suggest to call the function "isPodRunnable" to also capture the intention.

Reply:
sgtm